python/lsst/dax/imgserv/getimage/imagegetter_v1.py

```diff
@@ -0,0 +1,544 @@
+        return image
+
+
+    def _imagedataset_md(self):
+        # Return the butler policy name to retrieve metadata
+        return self._imagedataset_type + "_md"
+
+
+    def _cutout_from_src(self, src_image, xy_center_x, xy_center_y, width, height, wcs):
+        # Returns an image cutout from the source image.
+        #   srcImage - Source image.
+        #   xy_center - The center of region to cutout in pixels.
+        #   width - The width in pixels.
+        #   height - The height in pixels.
+        #   height and width will be trimmed if they go past the edge of the source image.
+        # First, center the cutout image
+        pix_ulx = int(xy_center_x - width/2.0)
+        pix_uly = int(xy_center_y - height/2.0)
+        xy_center = afw_geom.Point2I(pix_ulx, pix_uly)
+        log.debug("xy_center={}".format(xy_center))
+        src_box = src_image.getBBox()
+        # assuming both src_box and xy_center to be in Box2I
+        co_box = afw_geom.Box2I(xy_center, afw_geom.Extent2I(int(width), int(height)))
+        if co_box.overlaps(src_box):
+            co_box.clip(src_box)
+        else:
+            self._log.debug("cutout image wanted is OUTSIDE source image -> None")
+            return None
+        if isinstance(src_image, afw_image.ExposureF):
+            self._log.debug("co_box pix_ulx={} pix_end_x={} pix_uly={} pix_end_y={}"
+                  .format(pix_ulx, pix_ulx+width, pix_uly, pix_uly+height))
+            # img will keep wcs from source image
+            image = afw_image.ExposureF(src_image, co_box)
+        else:
+            # hack for non-ExposureF, e.g. raw (DecoratedImage)
+            pix_ulx = co_box.getBeginX() - src_image.getX0()
+            pix_end_x = co_box.getEndX() - src_image.getX0()
+            pix_uly = co_box.getBeginY() - src_image.getY0()
+            pix_end_y = co_box.getEndY() - src_image.getY0()
+            self._log.debug("co_box pix_ulx={} pix_end_x={} pix_uly={} pix_end_y={}".format(pix_ulx,
+                pix_end_x, pix_uly, pix_end_y))
+            image = src_image[pix_ulx:pix_end_x, pix_uly:pix_end_y].clone()
+        return image
+
+
+    def _data_id_from_qr(self, qresults):
+        # identify and fetch the data ID from 1 of 2 defined sets
+        ln = qresults[0]
+        # first try run, camcol, field, filter keys
+        if self._butler_keys == ["run", "camcol", "field", "filter"]:
+            run, camcol, field, filtername = ln[2:6]
+            data_id = { "run": run, "camcol": camcol, "field": field,
+                    "filter": filtername}
+            return "rcff", data_id
+        # if no match, then try tract, patch, filter keys
+        if self._butler_keys == ["tract", "patch", "filter"]:
+            tract, patch, filtername = ln[2:5]
+            data_id = { "tract": tract, "patch": patch, "filter":
+                    filtername }
+            return "tpf", data_id
+
+
+    def _keep_within_180(self, target, val):
```

Comment:
`normalize_positions`? This name isn't very descriptive.

Reply:
See prior comment - will remove all "tpf" and "rcff"  tagging.  If needed, data_id.keys() will be used to tell which is which.